bin/oneoff/compare_pardot_updates.rb

```diff
@@ -0,0 +1,234 @@
+# look reasonable.
+#
+# Specifically, we want to compare an older version of contact_rollups table in a reporting database
+# to its latest version in production database.
+# The changes will be written to contact_rollups_future table in the reporting database
+# for further analysis.
+#
+# This script can be modified to compare other tables.
+#
+# Analysis examples:
+# How many rows changed values in "country" column?
+#   select count(*) from contact_rollups_future where country is not null;
+#
+# What are those new values? (Note, 'null' string means the new value is NULL.)
+#   select country, count(*) as cnt from contact_rollups_future group by country order by cnt desc;
+#
+# What were the previous values?
+#   select a.country, b.country as new_country, count(*) as cnt
+#   from contact_rollups as a inner join contact_rollups_future as b on a.id = b.id
+#   where b.country is not null
+#   group by country, new_country
+#   order by cnt desc;
+
+require File.expand_path('../../../pegasus/src/env', __FILE__)
+require_relative '../../deployment'
+require src_dir 'database'
+require 'cdo/pegasus'
+
+# Production database has a global max query execution timeout setting.
+# This 30 minute setting can be used to override the timeout for a specific session or query.
+MAX_EXECUTION_TIME_SEC = 1800
+
+# Connection to read from Pegasus production database.
+PEGASUS_DB_READER = sequel_connect(
+  CDO.pegasus_db_reader,
+  CDO.pegasus_db_reader,
+  query_timeout: MAX_EXECUTION_TIME_SEC
+)
+
+# Connection to read from Pegasus reporting database.
+PEGASUS_REPORTING_DB_READER = sequel_connect(
+  CDO.pegasus_reporting_db_reader,
+  CDO.pegasus_reporting_db_reader,
+  query_timeout: MAX_EXECUTION_TIME_SEC
+)
+
+# Connection to write to Pegasus reporting database.
+PEGASUS_REPORTING_DB_WRITER = sequel_connect(
+  CDO.pegasus_reporting_db_writer,
+  CDO.pegasus_reporting_db_writer,
+  query_timeout: MAX_EXECUTION_TIME_SEC
+)
+
+# Table to write what data have changed to
+OUTPUT_TABLE = :contact_rollups_future
+
+# Ignore datetime columns such as updated_at, pardot_sync_at
+COLUMNS_TO_COMPARE = [
+  :id, :email, :opted_out, :dashboard_user_id, :name, :street_address, :city, :state, :country,
+  :postal_code, :district_name, :district_city, :district_state, :district_zip, :school_name,
+  :roles, :courses_facilitated, :professional_learning_enrolled, :professional_learning_attended,
+  :hoc_organizer_years, :grades_taught, :ages_taught, :email_malformed, :forms_submitted,
+  :form_roles, :opt_in
+]
+
+# Personal identifiable information can be used to compare but
+# will not be written out to output table.
+PII_COLUMNS = [:email, :name]
+
+def create_output_table
+  if PEGASUS_REPORTING_DB_WRITER.table_exists?(OUTPUT_TABLE)
+    p "#{OUTPUT_TABLE} table already exists"
+    PEGASUS_REPORTING_DB_WRITER.drop_table(OUTPUT_TABLE)
+    p "Dropped #{OUTPUT_TABLE} table"
+  end
```

Comment:
Is it important to report this behavior? If not, could just use [`create_table!`](https://github.com/jeremyevans/sequel/blob/master/doc/schema_modification.rdoc#create_table-1) below.

Reply:
No it isn't. create_table! is better.